### PR TITLE
Fix label inequality filtering

### PR DESCRIPTION
**Pull request by daviddavis**
[noissue]

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/tests/functional/api/using_plugin/test_labels.py [hunk 1]
```diff
@@ -174,9 +176,6 @@ def test_label_select(self):
         repos = self._filter_labels("environment")
         self.assertEqual(2, len(repos))
 
-        repos = self._filter_labels("!environment")
-        self.assertEqual(0, len(repos))
```
Comment (daviddavis) [inline pulpcore/tests/functional/api/using_plugin/test_labels.py:178]: If you have any repos in your system, this fails. Added a check below that is less prone to this.

File: pulpcore/tests/functional/api/using_plugin/test_labels.py [hunk 2]
```diff
@@ -164,6 +164,8 @@ def test_label_select(self):
         self._create_repo(labels)
         labels = {"environment": "staging", "certified": "false"}
         self._create_repo(labels)
+        labels = {}
+        self._create_repo(labels)
```
Comment (daviddavis) [inline pulpcore/tests/functional/api/using_plugin/test_labels.py:168]: This is the test here: this repo no longer gets returned for the `environment!=production` check below.

**Review by mdellweg — APPROVED** [review]
Looks good to Monkey :speak_no_evil:

**Review by goosemania — APPROVED** [review]

**Review by bmbouter — APPROVED** [review]
Thank you!

**Pull request merged**